feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt

```diff
@@ -137,21 +145,44 @@ sealed class StakeViewState<S>(
         }
     }
 
+    @ExperimentalCoroutinesApi
+    fun countdownTimerFlow(millis: Long) = callbackFlow<Long> {
+        val timer = object : CountDownTimer(millis, 1000) {
+            override fun onTick(millisUntilFinished: Long) {
+                offer(millisUntilFinished)
+            }
+
+            override fun onFinish() {
+                offer(0)
+
+                close()
+            }
+        }
+
+        timer.start()
+
+        awaitClose { timer.cancel() }
+    }.flowOn(Dispatchers.Main)
```

Comment:
I suggest using a consistent approach everywhere. Meaning, either create a timer in the ViewModel using Flow or in a view.

I personally leaning towards View approach via extension which I suggested above

Reply:
Done